core/ir/disassemble_shared.c

```diff
@@ -1164,6 +1224,48 @@ sign_extend_immed(instr_t *instr, int srcnum, opnd_t *src)
     }
 }
 
+#define MAX_LENGTH_ALL_CATEGORIES_AS_STRING 57
+
+/* Returns a space-separated string representation of the list of categories an
+ * instruction belongs to.
+ */
+void
+get_category_names(uint category, char category_names[], size_t category_names_size)
+{
+    memset(category_names, '\0', category_names_size);
```

Comment:
This seems inefficient: we just need a 0 at the end of the copy.  Normally we use standard helpers like print_to_buffer() or snprintf() + NULL_TERMINATE_BUFFER.

Reply:
Using `print_to_buffer()`, which takes care of that. Done.